### PR TITLE
Massmin choose collection

### LINKED ISSUE #5177: Mass Mint: Step1 - select collection
* #4964 

## Figma
---
- [Detailed interactions and logic is available Figma](https://www.figma.com/file/HCmIzbN0a2XhsOstHkLiwi/Mass-Mint---Handoff?node-id=0%3A1&t=ynDINS3WQGGG9Dj0-1). 
- all the needed screens
- [interactive prototype ](https://www.figma.com/proto/HCmIzbN0a2XhsOstHkLiwi/Mass-Mint---Handoff?page-id=0%3A1&node-id=1%3A332&viewport=1014%2C587%2C0.04&scaling=scale-down-width&starting-point-node-id=1%3A332&show-proto-sidebar=1)

Please take the prototype with a grain of salt, not everything is perfect and 1:1 to final design how it should look, so in case you dont know something, dont hesitate to ask @VladoTheBoi 

## STEP 1: SELECT COLLECTION
- user has to select a collection where he wants to mass mint NFTs
- dropdown shows all collections which user has
- last item in the list is "+ Create New Collection" which takes user to already existing and implemented Create Collection flow
  - its always displayed at the bottom 
- user also cannot interact with any other mass mint element, untill he selects a collection first
---

image
---
image
---
image
---
image

### PR DESCRIPTION
**Pull request by daiagi**
**Thank you for your contribution** to the [KodaDot NFT gallery](https://kodadot.xyz).

👇 \_ Let's make a quick check before the contribution.

## PR Type

- [ ] Bugfix
- [x] Feature
- [ ] Refactoring

## Context

- [x] Closes #5177 
- [x] Branch from #5217  
- [ ] Requires deployment <>

#### Before submitting pull request, please make sure:

- [x] My contribution builds **clean without any errors or warnings**
- [x] I've merged recent default branch -- **main** and I've no conflicts
- [x] I've tried to respect high code quality standards
- [x] I've didn't break any original functionality
- [x] I've posted a screenshot of demonstrated change in this PR

#### Optional

- [ ] I've tested it at </bsx/collection>
- [ ] I've tested PR on mobile
- [ ] I've written unit tests 🧪
- [ ] I've found edge cases

#### Had issue bounty label?

- [x] Fill up your KSM address: [Payout](https://beta.kodadot.xyz/transfer/?target=EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)

#### Community participation

- [ ] [Are you at KodaDot Discord?](https://discord.gg/35hzy2dXXh)

## Screenshot 📸

- [x] My fix has changed **something** on UI; a screenshot is best to understand changes for others.

image

### CODE REVIEW

File: components/massmint/Massmint.vue [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+      </NeoDropdown>
+    </section>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { NeoButton, NeoDropdown, NeoDropdownItem } from '@kodadot1/brick'
+import { OIcon } from '@oruga-ui/oruga'
+import { useMassmintsStore } from '@/stores/massmint'
+import TabItem from '@/components/explore/tab/TabItem.vue'
```
Comment (Jarsen136) [inline components/massmint/Massmint.vue:84]: How about moving `TabItem` component to `components/shared/TabItem.vue`

File: components/massmint/OnBoarding.vue [hunk 2]
```diff
@@ -0,0 +1,228 @@
...
+
+<script lang="ts" setup>
+import { NeoButton, NeoButtonVariant } from '@kodadot1/brick'
+import VueMarkdown from 'vue-markdown-render'
+import { useMassmintsStore } from '@/stores/massmint'
+import { SwipeDirection, useSwipe } from '@vueuse/core'
+const { $consola } = useNuxtApp()
+const router = useRouter()
+const { urlPrefix } = usePrefix()
+const { $i18n } = useNuxtApp()
```
Comment (Jarsen136) [inline components/massmint/OnBoarding.vue:74]:
```suggestion
const router = useRouter()
const { urlPrefix } = usePrefix()
const { $i18n, $consola } = useNuxtApp()
```

Comment (daiagi) [reply components/massmint/OnBoarding.vue:74]: Yes of course. Thank you

File: components/massmint/Massmint.vue (hunk 1 above)
Comment (daiagi) [reply components/massmint/Massmint.vue:84]: done

File: components/explore/tab/TabOnCollection.vue [hunk 3]
```diff
@@ -12,7 +12,7 @@
 </template>
 
 <script setup lang="ts">
-import TabItem from './TabItem.vue'
+import TabItem from '@/components/shared/TabItem.vue'
```
Comment (Jarsen136) [inline components/explore/tab/TabOnCollection.vue:15]:
```suggestion
```

File: components/explore/tab/TabOnExplore.vue [hunk 4]
```diff
@@ -12,7 +12,7 @@
 </template>
 
 <script setup lang="ts">
-import TabItem from './TabItem.vue'
+import TabItem from '@/components/shared/TabItem.vue'
```
Comment (Jarsen136) [inline components/explore/tab/TabOnExplore.vue:15]:
```suggestion
```

File: components/massmint/Massmint.vue [hunk 5]
```diff
@@ -0,0 +1,146 @@
...
+      </NeoDropdown>
+    </section>
+  </div>
+</template>
+
+<script setup lang="ts">
+import { NeoButton, NeoDropdown, NeoDropdownItem } from '@kodadot1/brick'
+import { OIcon } from '@oruga-ui/oruga'
+import { useMassmintsStore } from '@/stores/massmint'
+import TabItem from '@/components/shared/TabItem.vue'
```
Comment (Jarsen136) [inline components/massmint/Massmint.vue:84]:
```suggestion
```

We have auto import.

**Review by Jarsen136 — COMMENTED** [review]
small stuff

File: components/massmint/Massmint.vue [hunk 6]
```diff
@@ -0,0 +1,145 @@
+<template>
+  <div>
+    <section class="is-flex pb-6 is-flex-wrap-wrap row-gap">
+      <NeoButton class="mr-6" @click.native="toOnborading">
+        <o-icon pack="fas" icon="arrow-left" size="small" class="mr-1" />
```
Comment (roiLeo) [inline components/massmint/Massmint.vue:5]: Use icon prop from `<NeoButton>`?

**Review by roiLeo — CHANGES_REQUESTED** [review]
Something strange in dropdown hover state
Screenshot 2023-03-14 at 09-00-39 Low minting fees and carbonless NFTs KodaDot - NFT Market Explorer
Screenshot 2023-03-14 at 08-59-26 Low minting fees and carbonless NFTs KodaDot - NFT Market Explorer

- Valid checkbox should be rounded
- dont know how I managed to have this
Screenshot 2023-03-14 at 09-10-38 KodaDot - NFT Market Explorer

File: components/massmint/Massmint.vue (hunk 6 above)
Comment (daiagi) [reply components/massmint/Massmint.vue:5]: that puts the icon on the right

**Comment by daiagi** [comment]
> Something strange in dropdown hover state 

that's ```link-hover``` color from theme.
match it with the rest of the dropdown items?
>     * Valid checkbox should be rounded

do you have a collection with checkbox in it's name?


>     * dont know how I managed to have this
>       Screenshot 2023-03-14 at 09-10-38 KodaDot - NFT Market Explorer

ha! wtf?!

**Comment by roiLeo** [comment]
> ha! wtf?! on which page? which wallet? i'm unable to reproduce

On all page I'm blocked with this prefix..

**Comment by daiagi** [comment]
>Valid checkbox should be rounded
image


that is your collection's name. so nothing wrong here

**Comment by roiLeo** [comment]
> Valid checkbox should be rounded

compare it with figma
Capture d’écran 2023-03-14 à 9 57 00 AM

**Comment by daiagi** [comment]
>Valid checkbox should be rounded

Done

> dont know how I managed to have this

I'm sorry i have no idea how to reproduce that

the only place i have touched the url is ```components\navbar\createDropdown``` and it is strictly getting value, no set

**Review by roiLeo — COMMENTED** [review]
I don't have option to create collection from droplist when there is no collection
Screenshot 2023-03-14 at 10-30-37 Low minting fees and carbonless NFTs KodaDot - NFT Market Explorer


223053003-819695e3-334e-41be-bbce-ebcfe19026d8

side note: text alignment?

**Comment by daiagi** [comment]
> I don't have option to create collection from droplist when there is no collection

yes I'm not sure what to do with it. I guess in the meantime it can link to our current create collection page?

> 
> side note: text alignment?

looks good for me. ill check again cross browser

**Comment by daiagi** [comment]
* fixed text alignment between dropdown items
* fixed RMRK CollectionForMint query

File: queries/collectionForMint.graphql
Comment (vikiival) [inline queries/collectionForMint.graphql:1]: This is an old SubQuery file

Comment (vikiival) [reply queries/collectionForMint.graphql:1]: If resolve path points here something wrong is happening

Comment (daiagi) [reply queries/collectionForMint.graphql:1]: yeah it is outdated file

```ts
function resolveQueryPath(
  prefix: string,
  queryName: string
): Promise<typeof import('*.graphql')> {
  const path = getPath(prefix)
  return import(`@/queries/${path}${queryName}.graphql`)
}

function getPath(prefix: string) {
  switch (prefix) {
    case 'rmrk':
      return ''
    case 'bsx':
    case 'movr':
    case 'glmr':
    case 'snek':
    case 'subsquid':
    case 'rmrk2':
      return 'subsquid/general/'
    case 'statemine':
    case 'westmint':
      return 'unique/'
    case 'chain-bsx':
      return 'subsquid/bsx/'
    case 'chain-rmrk2':
      return 'subsquid/rmrk2/'
    default:
      return ''
  }
}
```


should ```rmrk``` point to ```subsquid/general``` as well?

Comment (daiagi) [reply queries/collectionForMint.graphql:1]: undid change to ```queries/collectionForMint.graphql```
use prefix instead

**Comment by daiagi** [comment]
>On all page I'm blocked with this prefix..

hey @roiLeo did you ever find out what the issue was / how to reproduce?
plus if you can review this please

**Comment by daiagi** [comment]
@Jarsen136 ready to roll?

**Review by Jarsen136 — APPROVED** [review]
lgtm

**Comment by Jarsen136** [comment]
> @Jarsen136 ready to roll?

Yup. Let it roll

**Comment by vikiival** [comment]
On chrome i have align


image

**Comment by exezbcz** [comment]
good job! 
image
please stick the footer to bottom, while keeping the collection dropdown at top

and then the create collection already mentioned - should redirect to collection creation 

unfortunately cannot test the ui when I select collection - just to be sure, this is then displayed? 
image

**Comment by daiagi** [comment]
Hey @exezbcz 
> please stick the footer to bottom, while keeping the collection dropdown at top

it will go down as more content is added to the page

> and then the create collection already mentioned - should redirect to collection creation

sure thing 

> unfortunately cannot test the ui when I select collection - just to be sure, this is then displayed?

not displayed yet, will be added in the next massmint PR's, for now it just the collection dropdown:
image
image
image
image
image
image

**Comment by exezbcz** [comment]
thanks! 
@daiagi the thing is that now there is black bg under the footer 
image

**Comment by daiagi** [comment]
> thanks! @daiagi the thing is that now there is black bg under the footer 

I can make the content section larger, but it will be temp css that will need to be removed later
it is a page still in progress

would you like me to create a UI shell here and make it functional in the rest of massmint issues?

**Review by vikiival — APPROVED** [review]

**Comment by yangwao** [comment]
😍 Perfect, I’ve sent the payout 
💵 $35 @ 34.87 USD/KSM ~ 1.004 $KSM
🧗 [EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L](https://kusama.subscan.io/account/EfmnRhHaQqfT3phm4cUCHCU3gFVDoSPR1U9WXzMRQBMqZ4L)
🔗 [0x2b32087e4db7e1763a9f07480bf8793665f29424560d345b7ea38db74c2a1291](https://kusama.subscan.io/extrinsic/0x2b32087e4db7e1763a9f07480bf8793665f29424560d345b7ea38db74c2a1291)

🪅 Let’s grab another issue and get rewarded! 
🪄 [github.com/kodadot/nft-gallery/issues](https://github.com/kodadot/nft-gallery/issues/)

**Pull request merged**